config/dynamic.c

```diff
@@ -16,11 +16,511 @@
+	int ret;
+	int conn_fd;
+	socklen_t len;
+	int rcv_buff_size;
+	struct sockaddr_un client_addr;
+
+	/* The lua state used to handle the dynamic configuration files. */
+	lua_State *lua_state;
+
+	len = sizeof(client_addr);
+	conn_fd = accept(server_socket_fd,
+		(struct sockaddr *)&client_addr, &len);
+        if (conn_fd < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to accept a new connection!\n");
+		return;
+	}
+
+	if (unlikely(client_addr.sun_family != AF_UNIX)) {
+		RTE_LOG(WARNING, GATEKEEPER,
+			"Unexpected condition: unknown client type %d at %s\n",
+			client_addr.sun_family, __func__);
+		goto close_fd;
+	}
+
+	/*
+	 * XXX The request must be received under a specified timeout,
+	 * or the request is aborted. This code needs further test.
+	 */
+	ret = setsockopt(conn_fd, SOL_SOCKET, SO_RCVTIMEO,
+		(const char*)&dy_conf->rcv_time_out, sizeof(struct timeval));
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to call setsockopt(SO_RCVTIMEO)\n");
+		goto close_fd;
+	}
+
+	rcv_buff_size = LUA_MSG_MAX_LEN;
+	ret = setsockopt(conn_fd, SOL_SOCKET,
+		SO_RCVBUF, &rcv_buff_size, sizeof(rcv_buff_size));
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to call setsockopt(SO_RCVBUF) with size = %d\n",
+			rcv_buff_size);
+		goto close_fd;
+	}
+
+	lua_state = luaL_newstate();
+	if (lua_state == NULL) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: failed to create new Lua state!\n");
+		goto close_fd;
+	}
+
+	/* Set up the Lua state while there is a connection. */
+	ret = setup_dy_lua(lua_state);
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to set up the lua state\n");
+		goto close_lua;
+	}
+
+	while (1) {
+		ret = process_single_cmd(conn_fd, lua_state);
+		if (ret != 0)
+			break;
+	}
+
+close_lua:
+	lua_close(lua_state);
+
+close_fd:
+	ret = close(conn_fd);
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to close the connection socket\n");
+	}
+}
+
+static int
+dyn_cfg_proc(void *arg)
+{
+	int ret = 0;
+	int sock_fd;
+	struct sockaddr_un server_addr;
+	uint32_t lcore = rte_lcore_id();
+	struct dynamic_config *dy_conf = arg;
+
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: the Dynamic Config block is running at lcore = %u\n", lcore);
+
+	/*
+	 * Remove any old socket and create an unnamed socket for the server.
+	 */
+	if(access(dy_conf->server_path, F_OK) != -1) {
+		ret = unlink(dy_conf->server_path);
+		if (ret != 0) {
+			exiting = true;
+			RTE_LOG(ERR, GATEKEEPER,
+				"dyn_cfg: Failed to unlink(%s)\n",
+				dy_conf->server_path);
+			goto out;
+		}
+	}
+
+	/* Init the server socket. */
+    	sock_fd = socket(AF_UNIX, SOCK_STREAM, 0);
+	if (sock_fd < 0) {
+		exiting = true;
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to initialize the server socket\n");
+		goto out;
+	}
+
+	/* Name the socket. */
+    	memset(&server_addr, 0, sizeof(server_addr));
+    	server_addr.sun_family = AF_UNIX;
+
+	if (sizeof(server_addr.sun_path) <= strlen(dy_conf->server_path)) {
```

Comment:
Good suggestion! I will keep the initialization code of the serve socket in ``run_dynamic_config()``.